ui/app/components/auth/form-template.ts

```diff
@@ -0,0 +1,210 @@
+    // renders tabs if listing visibility is set (auth tabs exist)
+    // and user has NOT clicked "Sign in with other"
+    if (this.authTabs && !this.showOtherMethods) {
+      return true;
+    }
+    return false;
+  }
+
+  @action
+  handleAuthSelect(element: string, event: HTMLElementEvent<HTMLInputElement> | null, idx: number) {
+    if (element === 'tab') {
+      this.setAuthTypeFromTab(idx);
+    } else if (event?.target?.value) {
+      this.selectedAuthMethod = event?.target.value;
+    }
+  }
+
+  setAuthTypeFromTab(idx: number) {
+    if (this.authTabs) {
+      this.selectedAuthMethod = Object.keys(this.authTabs)[idx] || '';
+      this.selectedTabIndex = idx;
+    }
+  }
+
+  @action
+  handleError(message: string) {
+    this.errorMessage = message;
+  }
+
+  @action
+  toggleView() {
+    this.showOtherMethods = !this.showOtherMethods;
+
+    if (this.renderTabs) {
+      // reset selected auth method to first tab
+      this.handleAuthSelect('tab', null, 0);
+    } else {
+      // all methods render, reset dropdown
+      this.selectedAuthMethod = 'token';
+    }
+  }
+
+  @action
+  async handleNamespaceUpdate(event: HTMLElementEvent<HTMLInputElement>) {
+    // update query param
+    this.args.handleNamespaceUpdate(event.target.value);
+    // reset tabs
+    this.authTabs = null;
+    // fetch mounts for that namespace
+    this.fetchMounts.perform(500);
+  }
+
+  fetchMounts = restartableTask(
+    waitFor(async (wait = 0) => {
+      // task is `restartable` so if the user starts typing again,
+      // it will cancel and restart from the beginning.
+      if (wait) await timeout(wait);
+
+      try {
+        // clear ember data store before re-requesting.. :(
+        this.store.unloadAll('auth-method');
+
+        const unauthMounts = await this.store.findAll('auth-method', {
+          adapterOptions: {
+            unauthenticated: true,
+          },
+        });
+
+        if (unauthMounts.length !== 0) {
+          this.authTabs = unauthMounts.reduce((obj: AuthTabs, m) => {
+            // serialize the ember data model into a regular ol' object
+            const mountData = m.serialize();
+            const methodType = mountData.type;
+            if (!Object.keys(obj).includes(methodType)) {
+              // create a new empty array for that type
+              obj[methodType] = [];
+            }
+
+            if (Array.isArray(obj[methodType])) {
+              // push mount data into corresponding type's array
+              obj[methodType].push(mountData);
+            }
+
+            return obj;
+          }, {});
+
+          // set tracked selected auth type to first tab
+          this.setAuthTypeFromTab(0);
+          // hide other methods to prioritize tabs (visible mounts)
+          this.showOtherMethods = false;
+        }
+      } catch (e) {
+        // if for some reason there's an error fetching mounts, swallow and just show standard form
+        this.authTabs = null;
+      }
+    })
```

Comment:
🫠 this fetch logic is essentially the same as here: https://github.com/hashicorp/vault/blob/cd0cd0a175ab6bc762df7bd5c2cc9f40b26d9402/ui/app/components/auth-form.js#L199-L227

I purposefully didn't refactor further because the removal of ember-data this logic will drastically improve and simplify this. Additionally, I'm making changes as incrementally as possible to avoid regressions.